lighthouse-core/lib/url-shim.js

```diff
@@ -172,12 +179,34 @@ class URLShim extends URL {
       return false;
     }
   }
+
+  /**
+   * @param {string} hostname Either a `new URL(url).hostname` or a `networkRequest.parsedUrl.host`
+   * @return {boolean}
+   */
+  static isLikeLocalhost(hostname) {
+    return SECURE_LOCALHOST_DOMAINS.includes(hostname);
+  }
+
+  /**
+   * @param {NetworkRequest['parsedURL']['scheme']} scheme
+   * @return {boolean}
+   */
+  static isSecureScheme(scheme) {
+    return SECURE_SCHEMES.includes(scheme);
+  }
+
+  /**
+   * @param {NetworkRequest['protocol']} devtoolsProtocol Has no colon, unlike `new URL(href).protocol`
+   * @return {boolean}
+   */
+  static isNonNetworkProtocol(devtoolsProtocol) {
```

Comment:
while we've awoken the beast....

scheme is the actual `new URL(href).protocol.slice(0, protocol.indexOf(':')` while protocol can be other weird values like `h2` or `http/0.9` and has this weird colon difference, should we rename to scheme? don't necessarily have to fix all usages but makes the `slice(0, indexOf)` usage a lot clearer I think

if we stick with protocol, link to 
https://source.chromium.org/chromium/chromium/src/+/master:content/browser/devtools/protocol/network_handler.cc;l=598-611;drc=56d4a9a9deb30be73adcee8737c73bcb2a5ab64f?q=f:devtools%20responseReceived%20-f:out%20language:cpp&ss=chromium and maybe check for a colon at the end?

Reply:
i stuck with calling the method 'protocol' but added lots of context to explain the debacle.

thx for finding that cs link!